lib/monitor.js

```diff
@@ -98,46 +93,70 @@ class Monitor extends Events.EventEmitter {
 
     start(callback) {
 
-        let reporterIndex = 0;
+        Items.serial(Object.keys(this.settings.reporters), (reporterName, next) => {
+
+            const streamsSpec = this.settings.reporters[reporterName];
+            const streamObjs = [];
+
+            Items.serial(streamsSpec, (spec, done) => {
+
+                // Already created stream
+                if (typeof spec.pipe === 'function') {
+                    streamObjs.push(spec);
+                    if (typeof spec.start === 'function') {
+                        let start = [];
+                        if (spec.start.args) {
+                            start = start.concat(spec.start.args);
+                        }
+                        start.push(done);
+                        return spec.start.apply(spec, start);
+                    }
+                    return done();
+                }
 
-        Items.serial(this.settings.reporters, (item, next) => {
+                let ctor = undefined;
+                let stream = undefined;
+                let ctorOpts;
 
-            let reporter;
+                if (typeof spec === 'string') {
+                    ctorOpts = {
+                        module: spec
+                    };
+                }
+                else {
+                    ctorOpts = spec.ctor;
+                }
 
-            // Use hint of reporter index
-            let errorHint = `reporter [${reporterIndex}]`;
-            reporterIndex++;
+                ctor = require(ctorOpts.module);
+                ctor = ctorOpts.name ? ctor[ctorOpts.name] : ctor;
+                Hoek.assert(typeof ctor === 'function', `Error in ${reporterName}. ${ctorOpts.module} must be a constructor function.`);
 
-            // If it has a reporter constructor, then create a new one, otherwise, assume it is
-            // a valid pre-constructed reporter
-            if (item.reporter) {
+                const ctorArgs = ctorOpts.args ? ctorOpts.args.slice(0) : [];
+                ctorArgs.unshift(null);
 
-                // If the supply a path or module node, try to load it
-                if (typeof item.reporter === 'string') {
-                    item.reporter = require(item.reporter);
-                }
+                ctor = ctor.bind.apply(ctor, ctorArgs);
```

Comment:
This whacky syntax is needed to create dynamic objects with dynamic arguments into the constructor. Some day when node supports `...` this can go away. Till then, this is it.